### DISCUSSION TITLE
Monitoring/Metrics avaialble for apicurio-registry product

### DISCUSSION THREAD

**Discussion by GrahamBourke** [Apicurio/apicurio-registry#3570] [Q&A] [answered]
Are there any exposed metrics (heath, usage, etc.) within the product?

Since Java, these could be JMX, or more recently [Micrometer ](https://micrometer.io/) framework based which would allow a prometheus endpoint to be exposed for scraping.

Could not find anything in documentation

**Comment by justinaslelys** [comment, marked as answer]
Hello,

Both health and metrics are exposed by default (or I simply do not remember enabling them).
[Here is the doc](https://github.com/Apicurio/apicurio-registry/blob/main/app/src/main/java/io/apicurio/registry/metrics/README.md) explaining what is there.

I think you are right that the endpoint path is not documented, but intuitively metrics can be found at `/metrics`, health on `/health/ready` and `/health/live`.

**Comment by Charlotte-br560** [comment]
Hey there! It's great to see your interest in monitoring metrics for Apicurio Registry. Currently, there isn't explicit documentation on exposed metrics within the product. However, given its Java-based nature, it's possible that metrics could be available through JMX or Micrometer framework. Integrating a Prometheus endpoint for scraping is definitely a promising avenue to explore. While it's not officially documented, reaching out to the community or exploring the codebase could provide insights into potential metrics availability.
